spyder/plugins/pylint/plugin.py

```diff
@@ -129,16 +130,68 @@ def on_main_menu_available(self):
         mainmenu.add_item_to_application_menu(
             pylint_act, menu_id=ApplicationMenus.Source)
 
+    @on_plugin_teardown(plugin=Plugins.Editor)
+    def on_editor_teardown(self):
+        widget = self.get_widget()
+        editor = self.get_plugin(Plugins.Editor)
+
+        # Connect to Editor
+        widget.sig_edit_goto_requested.disconnect(editor.load)
+        editor.sig_editor_focus_changed.disconnect(self._set_filename)
+
+        pylint_act = self.get_action(PylintActions.AnalyzeCurrentFile)
+
+        # TODO: use new API when editor has migrated
+        pylint_act.setVisible(False)
+        editor.pythonfile_dependent_actions.remove(pylint_act)
+
+    @on_plugin_teardown(plugin=Plugins.Preferences)
+    def on_preferences_teardown(self):
+        preferences = self.get_plugin(Plugins.Preferences)
+        preferences.deregister_plugin_preferences(self)
+
+    @on_plugin_teardown(plugin=Plugins.Projects)
+    def on_projects_teardown(self):
+        widget = self.get_widget()
+
+        # Connect to projects
+        projects = self.get_plugin(Plugins.Projects)
+        projects.sig_project_loaded.disconnect(self._set_project_dir)
+        projects.sig_project_closed.disconnect(self._unset_project_dir)
+
+        self._set_project_dir = None
+        self._unset_project_dir = None

```

Comment:
```suggestion
```

I think these lines are not necessary because they overwrite the methods defined below.